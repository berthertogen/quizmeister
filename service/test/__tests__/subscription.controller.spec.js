test("should create subscription on Post", async () => {
  await buildPostRequest("quiz", "", quizFactory.create([]));
  await buildPutRequest("quiz", "1/status/2", {});

  const response = await buildPostRequest("subscription", "", subscriptionFactory.create(1));
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(subscriptionFactory.ignorePropertiesEvent());
});

test("should get subscription on Get (email)", async () => {
  const response = await buildPostRequest('subscription', 'email', subscriptionFactory.userInput());
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(subscriptionFactory.ignoreProperties());
});

test("should get subscription on Get (quiz)", async () => {
  const response = await buildGetRequest('subscription', '1');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(subscriptionFactory.ignoreProperties());
});

test("should update subscription status Put (status)", async () => {
  await buildPutRequest("quiz", "1/status/4", {});

  const response = await buildPutRequest('subscription', '1/status/2');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(subscriptionFactory.ignorePropertiesEvent());
});

test("should delete subscription Delete", async () => {
  const responseDelete = await buildDeleteRequest('subscription', '1');
  assertContentTypeJsonAndStatus200(responseDelete);
  expect(responseDelete.body).toMatchSnapshot(subscriptionFactory.ignorePropertiesEvent());

  const responseGet = await buildGetRequest('subscription', '1');
  assertContentTypeJsonAndStatus200(responseGet);
  expect(responseGet.body).toMatchSnapshot();
});
