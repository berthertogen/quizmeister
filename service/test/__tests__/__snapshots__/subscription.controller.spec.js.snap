// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create subscription on Post 1`] = `
Object {
  "message": null,
  "status": 1,
  "subscription": Object {
    "email": "email",
    "modifiedOn": Any<String>,
    "quiz": Object {
      "date": "2100-01-31T23:00:00",
      "location": "location",
      "maxSubscriptions": 20,
      "modifiedOn": Any<String>,
      "quizId": 1,
      "remark": "remark",
      "rounds": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "status": 2,
      "subscriptions": null,
      "title": "title",
    },
    "remark": "remark",
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 1,
    "subscriptionId": 1,
    "team": "team",
  },
}
`;

exports[`should delete subscription Delete 1`] = `
Object {
  "message": null,
  "status": 1,
  "subscription": Object {
    "email": "email",
    "modifiedOn": Any<String>,
    "quiz": Object {
      "date": "2100-01-31T23:00:00",
      "location": "location",
      "maxSubscriptions": 20,
      "modifiedOn": Any<String>,
      "quizId": 1,
      "remark": "remark",
      "rounds": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "status": 4,
      "subscriptions": null,
      "title": "title",
    },
    "remark": "remark",
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 2,
    "subscriptionId": 1,
    "team": "team",
  },
}
`;

exports[`should delete subscription Delete 2`] = `Array []`;

exports[`should get subscription on Get (email) 1`] = `
Array [
  Object {
    "email": "email",
    "modifiedOn": Any<String>,
    "quiz": Object {
      "date": "2100-01-31T23:00:00",
      "location": "location",
      "maxSubscriptions": 20,
      "modifiedOn": Any<String>,
      "quizId": 1,
      "remark": "remark",
      "rounds": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "status": 2,
      "subscriptions": null,
      "title": "title",
    },
    "remark": "remark",
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 1,
    "subscriptionId": 1,
    "team": "team",
  },
]
`;

exports[`should get subscription on Get (quiz) 1`] = `
Array [
  Object {
    "email": "email",
    "modifiedOn": Any<String>,
    "quiz": Object {
      "date": "2100-01-31T23:00:00",
      "location": "location",
      "maxSubscriptions": 20,
      "modifiedOn": Any<String>,
      "quizId": 1,
      "remark": "remark",
      "rounds": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "status": 2,
      "subscriptions": null,
      "title": "title",
    },
    "remark": "remark",
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 1,
    "subscriptionId": 1,
    "team": "team",
  },
]
`;

exports[`should update subscription status Put (status) 1`] = `
Object {
  "message": null,
  "status": 1,
  "subscription": Object {
    "email": "email",
    "modifiedOn": Any<String>,
    "quiz": Object {
      "date": "2100-01-31T23:00:00",
      "location": "location",
      "maxSubscriptions": 20,
      "modifiedOn": Any<String>,
      "quizId": 1,
      "remark": "remark",
      "rounds": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "status": 4,
      "subscriptions": null,
      "title": "title",
    },
    "remark": "remark",
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 2,
    "subscriptionId": 1,
    "team": "team",
  },
}
`;
